### PR TITLE
Update docs for Dataset and DataLoader

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/advanced/own-models.rst [hunk 1]
```diff
@@ -198,37 +198,52 @@ the ``ip_config`` argument specifies a ip configuration file, which contains the
...
 
 .. code-block:: python
 
-    train_data = GSgnnNodeTrainData(config.graph_name,
-                                    config.part_config,
-                                    train_ntypes=config.target_ntype,
-                                    node_feat_field=node_feat_fields,
-                                    label_field=config.label_field)
+    train_data = GSgnnData(config.part_config,
+                           node_feat_field=node_feat_fields)
```
Comment (zhjwy9343) [inline docs/source/advanced/own-models.rst:206]: In the `hgt_nc.py` line #275, the `node_feat_field` has been deleted. So might remove it here as well.

File: docs/source/advanced/own-models.rst [hunk 2]
```diff
@@ -198,37 +198,52 @@ the ``ip_config`` argument specifies a ip configuration file, which contains the
...
 
-Arguments of this class include the partition configuration JSON file path, which are the outputs of the :ref:`Step 1 <step-1>`. The ``graph_name`` can be found in the JSON file.
+Arguments of this class include the partition configuration JSON file path, which are the outputs of the :ref:`Step 1 <step-1>`.
 
-The other values, the ``train_ntypes``, the ``label_field``, and the ``node_feat_field``, should be consistent with the values in the raw data :ref:`input configuration JSON <input-config>` defined in the :ref:`Step 1 <step-1>`. The ``train_ntypes`` is the ``node_type`` that has ``labels`` specified. The ``label_fields`` is the value specified in ``label_col`` of the ``train_ntype``. The ``node_feat_field`` is a dictionary, whose key is the values of ``node_type``, and value is the values of ``feature_name``.
+The ``node_feat_field``, should be consistent with the values in the raw data :ref:`input configuration JSON <input-config>` defined in the :ref:`Step 1 <step-1>`. The ``node_feat_field`` is a dictionary, whose keys are the values of ``node_type``, and values are the values of ``feature_name``.
```
Comment (zhjwy9343) [inline docs/source/advanced/own-models.rst:210]: We might move this line to the line #246, if deleted it in `GSgnnData` construction.

File: docs/source/advanced/own-models.rst [hunk 3]
```diff
@@ -198,37 +198,52 @@ the ``ip_config`` argument specifies a ip configuration file, which contains the
...
+    train_idxs = train_data.get_node_train_set(config.target_ntype)
     # Define the GraphStorm train dataloader
-    dataloader = GSgnnNodeDataLoader(train_data, train_data.train_idxs, fanout=config.fanout,
-                                     batch_size=config.batch_size, device=device, train_task=True)
+    dataloader = GSgnnNodeDataLoader(train_data,
+                                     train_idxs, fanout=config.fanout,
+                                     batch_size=config.batch_size,
+                                     node_feats=node_feat_fields,
+                                     label_field=config.label_field,
+                                     device=device, train_task=True)
```
Comment (zhjwy9343) [inline docs/source/advanced/own-models.rst:224]: In the `hgt_nc.py`, there is no `device=device` argument. Should we delete it here too?

File: docs/source/advanced/own-models.rst [hunk 4]
```diff
@@ -198,37 +198,52 @@ the ``ip_config`` argument specifies a ip configuration file, which contains the
...
-    eval_dataloader = GSgnnNodeDataLoader(train_data, train_data.val_idxs,fanout=config.fanout,
-                                          batch_size=config.eval_batch_size, device=device,
+    val_idxs = train_data.get_node_val_set(eval_ntype)
+    eval_dataloader = GSgnnNodeDataLoader(train_data,
+                                          val_idxs,
+                                          fanout=config.fanout,
+                                          batch_size=config.eval_batch_size,
+                                          node_feats=node_feat_fields,
+                                          label_field=config.label_field,
+                                          device=device,
```
Comment (zhjwy9343) [inline docs/source/advanced/own-models.rst:234]: ditto

File: docs/source/advanced/own-models.rst [hunk 5]
```diff
@@ -198,37 +198,52 @@ the ``ip_config`` argument specifies a ip configuration file, which contains the
...
     # Optional: Define the evaluation dataloader
-    test_dataloader = GSgnnNodeDataLoader(train_data, train_data.test_idxs,fanout=config.fanout,
-                                          batch_size=config.eval_batch_size, device=device,
+    test_idxs = train_data.get_node_test_set(eval_ntype)
+    test_dataloader = GSgnnNodeDataLoader(train_data,
+                                          test_idxs,
+                                          fanout=config.fanout,
+                                          batch_size=config.eval_batch_size,
+                                          node_feats=node_feat_fields,
+                                          label_field=config.label_field,device=device,
```
Comment (zhjwy9343) [inline docs/source/advanced/own-models.rst:243]: ditto

File: docs/source/advanced/own-models.rst (hunk 1 above)
Comment (classicsong) [reply docs/source/advanced/own-models.rst:206]: Updated

**Review by zhjwy9343 — APPROVED** [review]
LGTM

**Pull request merged**